### PR TITLE
Users can add comment to a particular gif

### PR DESCRIPTION
**Pull request by goodnessaig1**

### CODE REVIEW

File: server/controller/gifComment.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+     //    GIF COMMENT
+    static async createComment(req, res){
+      try {
+        const { comment } = req.body;
+        const { gifId } = req.params;
+        const createdAt = moment().format('YYYY-MM-DD ');
+        const createdBy = req.user.email;
+        const userId = req.user.userId;
+
+        const gif = await pool.query(`SELECT * FROM gifs WHERE gif_id = ${gifId}`);
```
Comment (courage173) [inline server/controller/gifComment.js]: use a parametised query

File: server/controller/gifComment.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+class gifController {
+
+     //    GIF COMMENT
+    static async createComment(req, res){
+      try {
+        const { comment } = req.body;
+        const { gifId } = req.params;
+        const createdAt = moment().format('YYYY-MM-DD ');
+        const createdBy = req.user.email;
```
Comment (courage173) [inline server/controller/gifComment.js]: why do you need createdBy? when you have userId

File: server/controller/gifComment.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        const userId = req.user.userId;
+
+        const gif = await pool.query(`SELECT * FROM gifs WHERE gif_id = ${gifId}`);
+        if (gif.rows.length === 0) {
+        return res.status(404).json({
+            status: 'error',
+            error: 'Gif with the specified ID NOT found',
+        });
+        }
+        const user = await pool.query(`SELECT * FROM users WHERE id = ${userId}`);
```
Comment (courage173) [inline server/controller/gifComment.js]: same for this 2 guys below. parametised query

File: server/middleware/admin.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+/* eslint-disable consistent-return */
+const admin = (req, res, next) => {
```
Comment (courage173) [inline server/middleware/admin.js]: rename this to admin

File: server/middleware/validation.js [hunk 5]
```diff
@@ -46,7 +46,26 @@ const logIn = (req, res, next) => {
     });
 }
 
+const comment = (req, res, next) => {
+    const validationRule = {
+        "comment": "required|string|max:180"
+    }
+    validator(req.body, validationRule, {}, (err, status) => {
+        if (!status) {
+            res.status(412)
```
Comment (courage173) [inline server/middleware/validation.js]: please appropriate error code => 400 and .json

File: server/migration/migration.js [hunk 6]
```diff
@@ -13,17 +13,30 @@ const createTable = pool.query(
...
         FOREIGN KEY (user_id) 
         REFERENCES users (id));
-        
+      
+    CREATE TABLE IF NOT EXISTS gif_comment (
+          gif_id INTEGER NOT NULL,
+          comment_id SERIAL PRIMARY KEY,
+          comments VARCHAR NOT NULL,        
+          created_at timestamp with time zone NOT NULL,
+          created_by VARCHAR NOT NULL,
```
Comment (courage173) [inline server/migration/migration.js]: we dont need created by and why is the gig id also a foreign key referencing the gif table

File: server/models/gifModel.js [hunk 7]
```diff
@@ -14,8 +14,20 @@ module.exports = () => {
             PRIMARY KEY (gif_id),
             FOREIGN KEY (user_id) 
             REFERENCES users (id)
-            )`
-        );
+            );
```
Comment (courage173) [inline server/models/gifModel.js:17]: why creating this in multiple places, i dont understand??

File: server/routes/gifRoute.js [hunk 8]
```diff
@@ -1,10 +1,17 @@
 const router = require("express").Router()
 const controller = require('../controller/gifController');
+const commentController = require('../controller/gifComment');
 const authorization = require("../middleware/authorization");
+const validation = require("../middleware/validation");
 
 
 
+//   GIF ROUTES FOR UPLOADING
 router.post('/', authorization , controller.createGif );
```
Comment (courage173) [inline server/routes/gifRoute.js]: why the spacing between them

File: server/test/admin.spec.js [hunk 9]
```diff
@@ -89,11 +89,11 @@ describe('ADMIN AND EMPLOYEE CREATE ACCOUNT AND SIGNUP ROUTES',()=>{
 describe('POST /auth/v1/login',()=>{
     it('It should login a user with a valid email and password', (done)=>{
             chai.request(server)
-            .post('/auth/v1/login', validInfo)
+            .post('/auth/v1/login', validation.logIn)
```
Comment (courage173) [inline server/test/admin.spec.js]: i don't understand why are you passing this middleware inside this file??

**Review by courage173 — CHANGES_REQUESTED** [review]

File: server/models/gifModel.js (hunk 7 above)
Comment (goodnessaig1) [reply server/models/gifModel.js:17]: It was with this tesing that i was able to discover why the models did not work before. Buh i have been able to create a table now without that migration. but now inside models. so i don't know if i should delete the migration folder now that i can use the models?

File: server/migration/migration.js (hunk 6 above)
Comment (goodnessaig1) [reply server/migration/migration.js]: Ok sir. I have removed the created_by across all routes

File: server/middleware/validation.js (hunk 5 above)
Comment (goodnessaig1) [reply server/middleware/validation.js]: ok sir. i have done that

File: server/middleware/admin.js (hunk 4 above)
Comment (goodnessaig1) [reply server/middleware/admin.js]: I have renamed it to isAdmin. I don't know if that's what is expected of me here sir?

Comment (courage173) [reply server/middleware/admin.js]: yes exactly

**Pull request merged**